### ISSUE TITLE
Add DJI Video Output as an Input

### ISSUE THREAD

**Issue by sebastienvg** [t-asano/tinyviewplus#28]
Yesterday the community made some tremendous progress and got the video out working of the USB-C port of the DJI Goggles https://github.com/fpv-wtf/voc-poc 

I think that would be just awesome to have the option to use DJI Goggles and/or analog with TinyViewPlus, I wish I was able to contribute to the code base but it's a little over my skillset at this time.

**Comment by potre32** [comment]
Yes, please work on this, I love your program but I switched to DJI and I can't use it

**Comment by t-asano** [comment]
Thank you for requesting. Tiny View Plus v0.9.31 supports user defined usb video capture device. It could be another rough solution.

![connection](https://user-images.githubusercontent.com/14119184/135732838-15879966-1eac-4d8d-ad34-e38b39da5095.jpg)

<img width="154" alt="スクリーンショット 2021-10-03 7 04 31" src="https://user-images.githubusercontent.com/14119184/135732907-d3055bc6-4d16-4a78-9eb6-ad3750440344.png">

```
<camera>
    <name>USB</name>
    <description>720p trim 4:3</description>
    <grab>
        <width>1280</width>
        <height>720</height>
    </grab>
    <crop>
        <x>160</x>
        <y>0</y>
        <width>960</width>
        <height>720</height>
    </crop>
    <draw>
        <aspectRatio>4:3</aspectRatio>
    </draw>
</camera>
```

This solution needs following equipments:
- USB HDMI capture device.
- DJI video output device.

[DJI Goggle]--USB--[PC1(voc-poc)]--HDMI--[Video Capture Device]--USB--[PC2(TinyViewPlus)]

**Comment by OscarPonsetti-OSCARNOVA** [comment]
Hello how can I put 1 pilot with hdZero and 3 pilot with analog? I have an hdmi card to usb and it work very good on obs but I don't know how have 4 mixed video

**Comment by nkozawa** [comment]
I wrote a Japanese article about HDZero video input to TinyViewPlus recently. I hope that Google translate helps you to understand.
[https://www.nkozawa.com/blog/archives/7761](https://www.nkozawa.com/blog/archives/7761)

**Comment by OscarPonsetti-OSCARNOVA** [comment]
ah ok so when I attached and hdzero source the program will take the settings in the fpv.xml file but if I also attach analog receiver which file the program use?

**Comment by t-asano** [comment]
@nkozawa nice blog!

@OscarPonsetti-OSCARNOVA fpv.xml is a setting file for "additional" video device. You can use default video device (analog receiver) and additional video device (defined by fpv.xml) at same time.

Default video device:
- device name : "USB2.0 PC CAMERA"
- supported device : Eachine ROTG01, ROTG02, etc.
- setting : hardcoded in TVP program

Additional video device:
- device name : user defined
- supported device : USB capture device
- setting : defined by fpv.xml

**Comment by OscarPonsetti-OSCARNOVA** [comment]
Ok which name I have to write?
![1](https://github.com/t-asano/tinyviewplus/assets/150076705/043b8456-6b64-4509-acd6-f1843f3db4b6)

**Comment by OscarPonsetti-OSCARNOVA** [comment]
if I write <name>USB3.0 Capture</name> the program will crash

**Comment by OscarPonsetti-OSCARNOVA** [comment]
here the file xml:
<camera>
    <name>USB3.0 Capture</name>
    <description>squeezed 16:9</description>
    <grab>
        <width>1280</width>
        <height>720</height>
    </grab>
    <crop>
        <x>0</x>
        <y>0</y>
        <width>704</width>
        <height>480</height>
    </crop>
    <draw>
        <aspectRatio>16:9</aspectRatio>
    </draw>
</camera>

**Comment by OscarPonsetti-OSCARNOVA** [comment]
@t-asano  what is the problem with the name here?

**Comment by t-asano** [comment]
No problem. I'm pareparing answer for you. please wait.

**Comment by OscarPonsetti-OSCARNOVA** [comment]
If I use obs with this device all work good but tinyviewer continue to crash with the name 
` <name>USB3.0 Capture</name>`

**Comment by t-asano** [comment]
I think \<name\> is correct. It might be a prolrem of resolution settings. Please try settings below.
[exaple_usb30capture.zip](https://github.com/user-attachments/files/15843836/exaple_usb30capture.zip)

- 1080p_crop.xml ... please try first
- 720p_crop.xml ... lower load, but your device may not support

please rename xml file to "fpv.xml".

![fpv](https://github.com/t-asano/tinyviewplus/assets/14119184/983723f2-2962-46c0-9ef2-3810ba3d45e4)

**Comment by OscarPonsetti-OSCARNOVA** [comment]
nope I try with this two file and nothing change the program continue to crash

**Comment by t-asano** [comment]
@OscarPonsetti-OSCARNOVA
Please give me more detailed information.

- Model name or URL of the hdmi capture card.
- Does anything appear on the screen when TVP crashes?
- Does TVP crash even if the hdmi capture card is disconnected?

**Comment by OscarPonsetti-OSCARNOVA** [comment]
Here is the link of my capture card: https://amzn.eu/d/1eenjq7
If I didn't plug the hdmi card the program didn't crash

**Comment by t-asano** [comment]
I couldn't get "DigitConvert" product in japanese market. However, I've got two similar products.

## GUERMOK HD Capture Card (Z29A-FBA-JP)

### Result: FAIL

![image](https://github.com/t-asano/tinyviewplus/assets/14119184/d698c8ec-9f5c-47f2-8b49-50034413d40f)

![guermok](https://github.com/t-asano/tinyviewplus/assets/14119184/17c8040d-656f-4a82-8f0d-cd6fcbc4e1d0)

- Device Name: USB3 Video
- TVP freezes at Camera Setup.
- https://www.amazon.co.jp/gp/product/B0963JG2X3

I expect that this device has a save problem as "DigiConvert". It may take a long time to resolve.

## UGREEN HD USB Video Capture Card (40189)

### Result: PASS (Image was displayed)

![image](https://github.com/t-asano/tinyviewplus/assets/14119184/e6a54e59-0179-4722-982a-29f0d0717cac)

![ugreen](https://github.com/t-asano/tinyviewplus/assets/14119184/e95323be-0f8f-47c3-818f-aa67c4c2ecef)

- Device Name: USB Video
- https://www.amazon.co.jp/gp/product/B0BGMYPBF4

**Comment by OscarPonsetti-OSCARNOVA** [comment]
Ok so  you will try to fix the problem or I need to buy the ugreen converter?

**Comment by t-asano** [comment]
If you are in a hurry, I recommend that you borrow a UGREEN device from someone to test or buy it yourself.

I can't take responsibility for the perfect operation of UGREEN devices. Because, I have only done limited testing, and I don't have enough time to do any more testing. Furthermore, there is no guarantee that the products that can be purchased in the Japanese market and your country have exactly the same specifications.

The problem with the "USB3" device may depend on the library (openFrameworks) used by TVP. I haven't investigated it yet, and I don't have enough time to do so. I can't even predict whether I can fix it.

**Comment by t-asano** [comment]
GUERMOK HD Capture Card (Z29A-FBA-JP) did not work even with apps below.

- Windows Camera
- OBS Studio (Windows)

> I expect that this device has a same problem as "DigiConvert". It may take a long time to resolve.

It seems like a different problem. The device I bought may be broken. I cannot investigate it.
